backend/app/assets/stylesheets/spree/backend/components/_navigation.scss

```diff
@@ -88,21 +171,11 @@ nav.menu {
 
       // flyout nav
       > ul {
-        position: absolute;
-        top: 0;
-        left: 100%;
-        width: $width-sidebar-flyout;
-        margin-left: 0;
-        border: $border-sidebar;
-        background: $color-navbar-submenu-bg;
-        box-shadow: $box-shadow;
-        @include caret($direction: left, $color-caret: $color-navbar-submenu-bg);
-
-      &:before {
-        z-index: 1;
-        top: 1.5em;
+        @include flyout;
+        @include media($tablet) {
+          display: none;
+        }
       }
-     }
     }
   }
```

Comment:
@davidedistefano I find this part is a bit confusing (probably it already was) and I'd take advantage of this change to make a small refactor:

```scss
.admin-nav-menu {
  ul {
    margin: 0;
    padding: 0;
    list-style: none;
  }

  li {
    background: $color-navbar-bg;

    &:hover {
      background: $color-navbar-active-bg;

      > a {
        color: $color-navbar-active;
      }
    }

    // flyout nav
    > ul {
      position: absolute;
      top: 0;
      left: 100%;
      width: $width-sidebar-flyout;
      margin-left: 0;
      border: $border-sidebar;
      background: $color-navbar-submenu-bg;
      box-shadow: $box-shadow;
      @include caret($direction: left, $color-caret: $color-navbar-submenu-bg);

      @include media($tablet) {
        position: relative;
        left: 0;
        width: $width-sidebar;
        border-top: 0;
        border-bottom: 0;
      }

      &:before {
        z-index: 1;
        top: 1.5em;
      }
    }

    &.selected {
      background: $color-navbar-active-bg;
      @include media($desktop) {
        position: relative;
      }

      &:not(:hover) > ul {
        @include media($desktop) {
          display: none;
        }
        @include media($tablet) {
          display: block;
        }
      }

      li {
        padding-left: 2em;
      }

      > a {
        color: $color-navbar-active;
      }
    }

    &:not(.selected) {
      position: relative;

      > ul {
        @include media($tablet) {
          display: none;
        }
      }

      &:not(:hover) > ul {
        display: none;
      }
    }
  }
```

Since the flyout stuff is used just once, we can remove the `flyout` mixin.